snapcraft/internal/parser.py

```diff
@@ -140,6 +144,73 @@ def _process_subparts(project_part, subparts, parts, origin, maintainer,
     return parts_list, after_parts
 
 
+def _process_entry(data):
+    key = data.get('project-part')
+    parts_list = {}
+    # Store all the parts listed in 'after' for each included part so that
+    # we can check later that we aren't missing any parts.
+    # XXX: What do we do about 'after' parts that should be looked for in
+    # the wiki?  They should be in the master parts list.
+    after_parts = set()
+
+    logger.debug('Processing part {!r}'.format(key))
+
+    try:
+        origin = data['origin']
+        origin_type = data.get('origin-type')
+        project_part = data['project-part']
+        subparts = data['parts']
+        maintainer = data['maintainer']
+        description = data['description']
+    except KeyError as e:
+        raise InvalidEntry("Missing key in wiki entry: {}".format(e))
+
+    # TODO: this should really be based on the origin uri not
+    # the part name to avoid the situation where there are multiple
+    # parts being pulled from the same repo branch.
+    origin_dir = os.path.join(BASE_DIR, key)
+    try:
+        os.makedirs(origin_dir)
+    except FileExistsError:
+        pass
+
+    class Options():
```

Comment:
Althought this is older code, it feels cheaper and more straightforward to just create the `Options` class with the proper attributes.

```python
class Options:
    source = origin
    source_type = None
```

Also note that `()` for the class declaration should not be required